src/HotChocolate/Language/src/Language.SyntaxTree/Utilities/SyntaxSerializer.QuerySyntax.cs

```diff
@@ -0,0 +1,224 @@
+using System;
+
+namespace HotChocolate.Language.Utilities
+{
+    public sealed partial class SyntaxSerializer
+    {
+        private void VisitOperationDefinition(
+            OperationDefinitionNode node,
+            ISyntaxWriter writer)
+        {
+            if (node.Name is { })
+            {
+                writer.Write(node.Operation.ToString().ToLowerInvariant());
+                writer.WriteSpace();
+
+                writer.WriteName(node.Name);
+                if (node.VariableDefinitions.Count > 0)
+                {
+                    writer.Write('(');
+
+                    writer.WriteMany(node.VariableDefinitions, VisitVariableDefinition);
+
+                    writer.Write(')');
+                }
+
+                writer.WriteMany(node.Directives,
+                    (n, w) => w.WriteDirective(n),
+                    w => w.WriteSpace());
+
+                writer.WriteSpace();
+            }
+            else if (node.Operation != OperationType.Query)
+            {
+                writer.Write(node.Operation.ToString().ToLowerInvariant());
+                writer.WriteSpace();
+            }
+
+            VisitSelectionSet(node.SelectionSet, writer);
+        }
+
+        private void VisitVariableDefinition(VariableDefinitionNode node, ISyntaxWriter writer)
+        {
+            writer.WriteVariable(node.Variable);
+
+            writer.Write(": ");
+
+            writer.WriteType(node.Type);
+
+            if (node.DefaultValue is { })
+            {
+                writer.Write(" = ");
+                writer.WriteValue(node.DefaultValue);
+            }
+        }
+
+        private void VisitFragmentDefinition(FragmentDefinitionNode node, ISyntaxWriter writer)
+        {
+            writer.Write(Keywords.Fragment);
+            writer.WriteSpace();
+
+            writer.WriteName(node.Name);
+            writer.WriteSpace();
+
+            if (node.VariableDefinitions.Count > 0)
+            {
+                writer.Write('(');
+
+                writer.WriteMany(
+                    node.VariableDefinitions,
+                    VisitVariableDefinition);
+
+                writer.Write(')');
+                writer.WriteSpace();
+            }
+
+            writer.Write(Keywords.On);
+            writer.WriteSpace();
+
+            writer.WriteNamedType(node.TypeCondition);
+
+            writer.WriteMany(node.Directives,
+                (n, w) => w.WriteDirective(n));
+
+            if (node.SelectionSet is { })
+            {
+                writer.WriteSpace();
+                VisitSelectionSet(node.SelectionSet, writer);
+            }
+        }
+
+        private void VisitSelectionSet(SelectionSetNode node, ISyntaxWriter writer)
+        {
+            if (node is { } && node.Selections.Count > 0)
```

Comment:
If we enable nullable reference types, should we check for null in private methods?

Reply:
no, this condition is wrong :)